### PR TITLE
helm chart to install the operator

### PR DESCRIPTION
**Pull request by cmendible**
Adding helm chart to install the k8spin-operator.

Fixing #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,23 @@ into Tenant resources.
 
 Clone this repo, cd into it and:
 
+### Install with Helm 3
+
+```bash
+# Create a local cluster
+$ kind create cluster
+# Deploy cert-manager
+$ helm repo add jetstack https://charts.jetstack.io
```
Comment (angelbarrera92) [inline README.md:38]: jetstack?

File: charts/k8spin-operator/templates/k8spin-webhook.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: default/k8spin-webhook-cert
```
Comment (angelbarrera92) [inline charts/k8spin-operator/templates/k8spin-webhook.yaml:5]: Replace `default` to `{{ .Release.Namespace }}`

File: charts/k8spin-operator/templates/k8spin-webhook.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          - "v1"
+        operations:
+          - CREATE
+          - UPDATE
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: default/k8spin-webhook-cert
```
Comment (angelbarrera92) [inline charts/k8spin-operator/templates/k8spin-webhook.yaml:48]: Replace `default` to `{{ .Release.Namespace }}`

File: charts/k8spin-operator/templates/k8spin-webhook.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      caBundle: Cg==
+    rules:
+      - apiGroups: ["k8spin.cloud"]
+        resources:
+          - "spaces"
+        apiVersions:
+          - "v1"
+        operations:
+          - CREATE
+          - UPDATE
```
Comment (angelbarrera92) [inline charts/k8spin-operator/templates/k8spin-webhook.yaml:102]: Add a new line at the end of every file :)

File: charts/k8spin-operator/templates/k8spin_webhook_certificate.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: k8spin-webhook-cert
+  namespace: {{ .Release.Namespace }}
+spec:
+  dnsNames:
+  - k8spin-webhook.default.svc
```
Comment (angelbarrera92) [inline charts/k8spin-operator/templates/k8spin_webhook_certificate.yaml:8]: Replace `default` to `{{ .Release.Namespace }}`

**Comment by angelbarrera92** [comment]
Looks cool for the first iteration of this helm chart. 
What is missing is (maybe some of these points are out of the scope of the PR):

- Automatic upload to the GitHub pkg registry (as mentioned in #4 using the `export HELM_EXPERIMENTAL_OCI=1`) on every release. GitHub action
- Follow all the best practices of creating a helm chart:
  - https://helm.sh/docs/chart_best_practices/labels/ (missing)
  - https://helm.sh/docs/chart_best_practices/pods/ (missing)
  - https://helm.sh/docs/chart_best_practices/rbac/ (missing)
- Consider adding cert-manager helm chart as a dependency. 
  - https://helm.sh/docs/chart_best_practices/dependencies/
  - Not sure how it works, maybe with something like: `condition: cert-manager.enabled` in the k8spin-operator values.
- Helm chart documentation (automatic generated)
  - https://medium.com/rapids-ai/introducing-frigate-a-documentation-generation-tool-for-kubernetes-1791854031a1#cid=av01_so-twit_en-us

Let us know if you want to continue working on it, or we can continue with this 🔝 stuff!

Thanks a lot @cmendible!

**Comment by cmendible** [comment]
Hi @angelbarrera92 can we create a diff issue for the Automatic upload to the GitHub pkg registry?

Regarding the cert-manager dependency I did add it in my first iteration but it does not work as expected. I will also separate this and explore it once we have a first release of the chart.

I'll work this week on the best-practices:
* https://helm.sh/docs/chart_best_practices/labels/ 
* https://helm.sh/docs/chart_best_practices/pods/ 
* https://helm.sh/docs/chart_best_practices/rbac/

**Comment by cmendible** [comment]
@angelbarrera92 please check the chart. Best practices are in place and the readme file was generated with frigate.

**Comment by angelbarrera92** [comment]
> Hi @angelbarrera92 can we create a diff issue for the Automatic upload to the GitHub pkg registry?
> 
> Regarding the cert-manager dependency I did add it in my first iteration but it does not work as expected. I will also separate this and explore it once we have the first release of the chart.
> 
> I'll work this week on the best-practices:
> 
>     * https://helm.sh/docs/chart_best_practices/labels/
> 
>     * https://helm.sh/docs/chart_best_practices/pods/
> 
>     * https://helm.sh/docs/chart_best_practices/rbac/

Ye! I think that's the honest way to do it. We will create more Issues to enrich this awesome helm chart!

> @angelbarrera92 please check the chart. Best practices are in place and the readme file was generated with frigate.

Thanks for the effort to contribute to this project!

**Review by angelbarrera92 — APPROVED** [review]
LGTM! 👏👏👏👏👏👏

**Review by paurosello — APPROVED** [review]

**Comment by paurosello** [comment]
thanks for your contribution @cmendible !

**Pull request merged**